vulkan/vulkan_application.cc

```diff
@@ -0,0 +1,156 @@
+// Copyright 2016 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "flutter/vulkan/vulkan_application.h"
+#include <utility>
+#include <vector>
+#include "flutter/vulkan/vulkan_device.h"
+#include "flutter/vulkan/vulkan_proc_table.h"
+#include "flutter/vulkan/vulkan_utilities.h"
+
+namespace vulkan {
+
+VulkanApplication::VulkanApplication(
+    VulkanProcTable& p_vk,
+    const std::string& application_name,
+    std::vector<std::string> enabled_extensions,
+    uint32_t application_version,
+    uint32_t api_version)
+    : vk(p_vk), api_version_(api_version), valid_(false) {
+  // Check if we want to enable debugging.
+
+  bool enable_instance_debugging =
+      EnableDebugging() && VulkanDebugReport::DebugExtensionSupported(vk);
+
+  // Configure extensions.
+
+  if (enable_instance_debugging) {
+    enabled_extensions.emplace_back(VulkanDebugReport::DebugExtensionName());
+  }
+
+  const char* extensions[enabled_extensions.size()];
```

Comment:
Lives on stack. Created as an `alloca`.

Compiling the follow program as checking bitcode [shows the `alloca`](https://gist.github.com/chinmaygarde/ba33bd7fed22c80c44e7e43fe14e07ed#file-main-bc-L18):
```
int main(int argc, char const *argv[]) {
  char b[argc];
  return 0;
}
```